src/routes/(home)/+page.server.ts

```diff
@@ -8,8 +10,28 @@ export async function load() {
 		.sort((a, b) => new Date(b.meta.date).getTime() - new Date(a.meta.date).getTime())
 		.slice(0, 6);
 
+	const baseURL = 'https://api.github.com/repos/torrust/';
+	const token = import.meta.env.VITE_GITHUB_TOKEN;
```

Comment:
Hi, @grmbyrn, since the app is hosted statically on GitHub Pages, the environment variable's value (`VITE_GITHUB_TOKEN`) would be embedded directly into the build output (JavaScript code).

This means anyone can view it by inspecting the network requests or source code of your hosted app.

On the other hand, I think we don't need to do it every time a user loads the homepage. For now, we can build the data at build time because we are making many changes. We could create a file like `src/lib/constants/constants.ts` or maybe add a new static JSON file with the info. We need to find the easiest way to do it.

In the future, it the page is not updated frequently we can add a workflow to update that list (rebuild and publish) preiodically.

Reply:
Hi @grmbyrn, as we discussed yesterday in the meeting, we can remove the token or use it only when it's provided (development environment). I did not realize the API rate limit is per IP, so it would be acceptable to do it at runtime. As @da2ce7 suggested, we can use a cache to store the result per user. I guess the cache could be implemented with https://svelte.dev/docs/svelte/stores.

Reply:
> Hi @grmbyrn, as we discussed yesterday in the meeting, we can remove the token or use it only when it's provided (development environment). I did not realize the API rate limit is per IP, so it would be acceptable to do it at runtime. As @da2ce7 suggested, we can use a cache to store the result per user. I guess the cache could be implemented with https://svelte.dev/docs/svelte/stores.

Hi @josecelano I was able to cache the data coming from the GitHub API with a store. The initial request is within the 60 requests an hour permitted by the API. After that it fetches the requests from the store. At the moment the cache expires every hour.